### ISSUE TITLE
Prevent name clashes in Components 

### ISSUE THREAD

**Issue by jrudolph** [lightbend/akkaserverless-java-sdk#711]
When you have several components that have the same name (but a different package), names may clash in `Components`.

**Comment by raboof** [comment]
Do you mean in for example function names that are derived from the component names? Perhaps we could generalize/improve https://github.com/lightbend/akkaserverless-java-sdk/blob/main/codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ComponentSourceGenerator.scala#L49-L55 ? Or does that also has the problem?

**Comment by jrudolph** [comment]
> Do you mean in for example function names that are derived from the component names?

Yes, I think it was the method names in the `Components` trait that had that problem.

> Perhaps we could generalize/improve [`main`/codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ComponentSourceGenerator.scala#L49-L55](https://github.com/lightbend/akkaserverless-java-sdk/blob/main/codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ComponentSourceGenerator.scala#L49-L55)

Cool, didn't know we have such a functionality. Where does that help in contrast?

**Comment by johanandren** [comment]
`name` here is from the `CallableComponent` which gets a unique one if there are several with the same name:
https://github.com/lightbend/akkaserverless-java-sdk/blob/main/codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ComponentSourceGenerator.scala#L73

So I thought I already solved this, that was a part of the reason for that logic at least (and I saw conflicts in the codegen compile tests without it).

**Comment by jrudolph** [comment]
Ok, I'm going to try to reproduce it again on Thursday.

**Comment by johanandren** [comment]
One name clash I didn't think of when we discussed this has not occurred, one of the package names in the Scala QuickStart `customer` now clashes with the generated `Components#customer` so that when the `ComponentsImpl` refers to the FQCN `customer.api.Customer` the Scala compiler thinks that it is a path dependent type from the `customer` getter. :/
